docusaurus/src/components/RequestERD/EmailModal.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from "react";
+import styles from "./EmailModal.module.css";
+import { faXmark } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import cn from "classnames";
+import { useDebouncedCallback } from "use-debounce";
+import useDocusaurusContext from "@docusaurus/useDocusaurusContext";
+
+const CloseButton = ({ onClose }) => {
+  return (
+    <button onClick={onClose} className={styles.modalContentClose}>
+      <FontAwesomeIcon icon={faXmark} />
+    </button>
+  );
+};
+
+const ModalBody = ({ status, onSubmit }) => {
+  const [email, setEmail] = useState("");
+  const [emailError, setEmailError] = useState("");
+
+  const debouncedValidation = useDebouncedCallback((value) => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    if (emailRegex.test(value)) {
+      setEmailError("");
+    } else {
+      setEmailError("Please enter a valid email address");
+    }
+  }, 800);
+
+  const handleOnChange = (e) => {
+    const emailValue = e.target.value;
+    setEmail(emailValue);
+    debouncedValidation(emailValue);
+  };
+
+  if (status === "success") {
+    return (
+      <div>
+        <p> Request submitted successfully!</p>
+        <p> We will notify you when the ERD is ready</p>
+      </div>
+    );
+  }
+  if (status === "error") {
+    return (
+      <div>
+        <p> Error submitting request. Please try again later.</p>
+      </div>
+    );
+  }
```

Comment:
When I submitted with an invalid email, the modal showed me this error, and even after closing the modal and clicking the `+ Request Entity-Relationship Diagram` button again, the modal opened in this error state rather than letting me input a new email and try again.

I could only get it out of this state by refreshing the page.

Reply:
Nice catch. I'm resetting `status` after closing the modal